src/main/java/org/betonquest/betonquest/objectives/BlockObjective.java

```diff
@@ -78,4 +106,31 @@ public void stop() {
         HandlerList.unregisterAll(this);
     }
 
+    private boolean checkLocation(final Location loc, final Profile profile) {
+        try {
+            if (location != null) {
+                if (location2 != null) {
+                    return isInRange(loc, profile);
+                }
+                return loc.equals(location.getLocation(profile));
+            }
+        } catch (QuestRuntimeException e) {
+            logger.error(instruction.getPackage(), e.getMessage());
+        }
+        return false;
+    }
+
+    private boolean isInRange(final Location loc, final Profile profile) throws QuestRuntimeException {
+        final Location loc1 = location.getLocation(profile);
+        final Location loc2 = location2.getLocation(profile);
+        return inBetween(loc1, loc2, loc);
+    }
+
+    private boolean inBetween(final Location range1, final Location range2, final Location pos) {
+        final boolean inX = Integer.min(range1.getBlockX(), range2.getBlockX()) <= pos.getBlockX() && pos.getBlockX() <= Integer.max(range1.getBlockX(), range2.getBlockX());
+        final boolean inY = Integer.min(range1.getBlockY(), range2.getBlockY()) <= pos.getBlockY() && pos.getBlockY() <= Integer.max(range1.getBlockY(), range2.getBlockY());
+        final boolean inZ = Integer.min(range1.getBlockZ(), range2.getBlockZ()) <= pos.getBlockZ() && pos.getBlockZ() <= Integer.max(range1.getBlockZ(), range2.getBlockZ());
+        final boolean inWorld = range1.getWorld().equals(range2.getWorld()) && range2.getWorld().equals(pos.getWorld());
+        return inX && inY && inZ && inWorld;
```

Comment:
I would put the conditions inline, so w
that we can profit from short-circuit evaluation.
It might be worth it to extract the inX/inY/inZ logic into its own method.